### PR TITLE
Fix access key value containing `=` edge case

### LINKED ISSUE #529: Bug: error: Could not parse secret for AWS Secret key
### Describe bug

openshift-adp-controller-manager pod (oadp 0.5.3) is reporting the following error :

```
2022-01-11T10:59:52.630Z        ERROR   controller.dataprotectionapplication    Reconciler error        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp", "error": "secret parsing error"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /go/src/github.com/openshift/oadp-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
2022-01-11T11:02:36.470Z        INFO    controller.dataprotectionapplication    got provider secret name: cloud-credentials     {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:02:36.471Z        INFO    controller.dataprotectionapplication    got provider secret name: cloud-credentials     {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:02:36.471Z        INFO    controller.dataprotectionapplication    got provider secret name: cloud-credentials     {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:02:36.471Z        INFO    controller.dataprotectionapplication    Could not parse secret for AWS Secret key       {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:02:36.471Z        INFO    controller.dataprotectionapplication    Error parsing provider secret cloud-credentials for backupstoragelocation openshift-adp/dpa-1        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:02:36.471Z        INFO    controller.dataprotectionapplication    Error building registry container for backupstoragelocation openshift-adp/dpa-1, could not fetch registry env vars   {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:02:36.481Z        ERROR   controller.dataprotectionapplication    Reconciler error        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp", "error": "secret parsing error"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /go/src/github.com/openshift/oadp-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
```
```
oc get secret cloud-credentials -oyaml
apiVersion: v1
data:
  cloud: <base64-code>
kind: Secret
metadata:
  creationTimestamp: "2022-01-11T11:07:28Z"
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:data:
        .: {}
        f:cloud: {}
      f:type: {}
    manager: kubectl-create
    operation: Update
    time: "2022-01-11T11:07:28Z"
  name: cloud-credentials
  namespace: openshift-adp
  resourceVersion: "179259626"
  selfLink: /api/v1/namespaces/openshift-adp/secrets/cloud-credentials
  uid: uid
type: Opaque
```
 ```
cat cloud-credentials
[default]
aws_access_key_id=<id>
aws_secret_access_key=<key>
```
```
oc logs oadp-dpa-1-aws-registry-pod
time="2022-01-11T10:10:16.165248929Z" level=debug msg="s3aws.Walk(\"/docker/registry/v2/repositories\")" go.version=go1.17.2 http.request.host="172.21.2.33:5000" http.request.id=c089c6bc-406b-497d-b1ce-441c00ca34e1 http.request.method=GET http.request.remoteaddr="172.21.2.1:36652" http.request.uri="/v2/_catalog?n=5" http.request.useragent=kube-probe/1.19 trace.duration=11.424021ms trace.file=/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).Walk" trace.id=d27dc568-5013-43b8-b413-1b709b8399f2 trace.line=232
time="2022-01-11T10:10:17.846306218Z" level=debug msg="authorizing request" go.version=go1.17.2 http.request.host="172.21.2.33:5000" http.request.id=40dace6d-0904-4d1f-a3e2-cee2bea0bd16 http.request.method=GET http.request.remoteaddr="172.21.2.1:36668" http.request.uri="/v2/_catalog?n=5" http.request.useragent=kube-probe/1.19
172.21.2.1 - - [11/Jan/2022:10:10:17 +0000] "GET /v2/_catalog?n=5 HTTP/1.1" 200 20 "" "kube-probe/1.19"
time="2022-01-11T10:10:17.854471409Z" level=debug msg="s3aws.ListObjectsV2Pages(docker/registry/v2/repositories/)" go.version=go1.17.2 http.request.host="172.21.2.33:5000" http.request.id=40dace6d-0904-4d1f-a3e2-cee2bea0bd16 http.request.method=GET http.request.remoteaddr="172.21.2.1:36668" http.request.uri="/v2/_catalog?n=5" http.request.useragent=kube-probe/1.19 trace.duration=8.062034ms trace.file=/go/src/github.com/docker/distribution/registry/storage/driver/s3-aws/s3.go trace.func="github.com/docker/distribution/registry/storage/driver/s3-aws.(*driver).doWalk" trace.id=f49b9978-928f-4fae-8a0b-4ef48b850306 trace.line=969 trace.parent.id=72166b2d-0a0c-47ca-aef7-b35204cdb843
time="2022-01-11T10:10:17.85453263Z" level=debug msg="s3aws.Walk(\"/docker/registry/v2/repositories\")" go.version=go1.17.2 http.request.host="172.21.2.33:5000" http.request.id=40dace6d-0904-4d1f-a3e2-cee2bea0bd16 http.request.method=GET http.request.remoteaddr="172.21.2.1:36668" http.request.uri="/v2/_catalog?n=5" http.request.useragent=kube-probe/1.19 trace.duration=8.147916ms trace.file=/go/src/github.com/docker/distribution/registry/storage/driver/base/base.go trace.func="github.com/docker/distribution/registry/storage/driver/base.(*Base).Walk" trace.id=72166b2d-0a0c-47ca-aef7-b35204cdb843 trace.line=232
time="2022-01-11T10:10:21.153508362Z" level=debug msg="authorizing request" go.version=go1.17.2 http.request.host="172.21.2.33:5000" http.request.id=8b367f2a-eb4d-4aad-a150-41865ee35fdf http.request.method=GET http.request.remoteaddr="172.21.2.1:36680" http.request.uri="/v2/_catalog?n=5" http.request.useragent=kube-probe/1.19
172.21.2.1 - - [11/Jan/2022:10:10:21 +0000] "GET /v2/_catalog?n=5 HTTP/1.1" 200 20 "" "kube-probe/1.19"
time="2022-01-11T10:10:21.161312452Z" level=debug msg="s3aws.ListObjectsV2Pages(docker/registry/v2/repositories/)" go.version=go1.17.2 http.request.host="172.21.2.33:5000" http.request.id=8b367f2a-eb4d-4aad-a150-41865ee35fdf http.request.method=GET http.request.remoteaddr="172.21.2.1:36680" http.request.uri="/v2/_catalog?n=5" http.request.useragent=kube-probe/1.19 trace.duration=7.706857ms trace.file=/go/src/github.com/docker/distribution/registry/storage/driver/s3-aws/s3.go trace.func="github.com/docker/distribution/registry/storage/driver/s3-aws.(*driver).doWalk" trace.id=53807c17-ea8b-4fb1-8989-01ad1181c6c0 trace.line=969 trace.parent.id=5aa09dc2-e1d2-4c73-93c4-6fff28b0cfe9
```

### What happened?

The following errors :
 "Error parsing provider secret cloud-credentials"
 "Could not parse secret for AWS Secret key"

### OADP Version

0.5.x (Stable)

### OpenShift Version

4.6

### Velero pod logs

```shell
pis/template.openshift.io/v1?timeout=32s
I0111 06:01:39.036300       1 request.go:665] Waited for 1.043526929s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/packages.operators.coreos.com/v1?timeout=32s
time="2022-01-11T06:06:26Z" level=error msg="Error backing up item" backup=openshift-adp/openshift error="error executing custom action (groupResource=imagestreams.image.openshift.io, namespace=openshift, name=app-example): rpc error: code = Unknown desc = Error trying to reuse blob sha256:f34b00c7da207ce777a45a37b1241d61bd613c43e3f7c017ceea736bc919c365 at destination: error pinging docker registry oadp-dpa-1-aws-registry-route-openshift-adp.apps.domain.com: Get \"http://oadp-dpa-1-aws-registry-route-openshift-adp.apps.domain.com/v2/\": dial tcp IP-ADDRESS:80: i/o timeout" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331" error.function="github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions" logSource="pkg/backup/backup.go:435" name=app-example
```

### Restic pod logs

```shell
no restic pods
```

### Operator pod logs

```shell
2022-01-11T11:08:04.162Z        INFO    controller.dataprotectionapplication    Error parsing provider secret cloud-credentials for backupstoragelocation openshift-adp/dpa-1        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:08:04.162Z        INFO    controller.dataprotectionapplication    Error building registry container for backupstoragelocation openshift-adp/dpa-1, could not fetch registry env vars   {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:08:04.205Z        ERROR   controller.dataprotectionapplication    Reconciler error        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp", "error": "secret parsing error"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /go/src/github.com/openshift/oadp-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
2022-01-11T11:18:59.571Z        INFO    controller.dataprotectionapplication    got provider secret name: cloud-credentials     {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:18:59.572Z        INFO    controller.dataprotectionapplication    got provider secret name: cloud-credentials     {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:18:59.572Z        INFO    controller.dataprotectionapplication    got provider secret name: cloud-credentials     {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:18:59.572Z        INFO    controller.dataprotectionapplication    Could not parse secret for AWS Secret key       {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:18:59.572Z        INFO    controller.dataprotectionapplication    Error parsing provider secret cloud-credentials for backupstoragelocation openshift-adp/dpa-1        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:18:59.572Z        INFO    controller.dataprotectionapplication    Error building registry container for backupstoragelocation openshift-adp/dpa-1, could not fetch registry env vars   {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp"}
2022-01-11T11:18:59.584Z        ERROR   controller.dataprotectionapplication    Reconciler error        {"reconciler group": "oadp.openshift.io", "reconciler kind": "DataProtectionApplication", "name": "dpa", "namespace": "openshift-adp", "error": "secret parsing error"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
        /go/src/github.com/openshift/oadp-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
```

### New issue

- [X] This issue is new

OADP-378

### PR DESCRIPTION
**Pull request by kaovilai**
Close #529 
[OADP-378](https://issues.redhat.com/browse/OADP-378)

PR CatalogSource
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: oadp-secretparsing
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: ghcr.io/kaovilai/oadp-operator-catalog:secretparsing
  displayName: Tiger's OADP SecretParsing
  publisher: Tiger Kaovilai
  updateStrategy:
    registryPoll:
      interval: 10m
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,9 +48,9 @@ ENVTEST_K8S_VERSION = 1.21
 
 .PHONY:ginkgo
 ginkgo: # Make sure ginkgo is in $GOPATH/bin
-	go get github.com/onsi/ginkgo/ginkgo
-	go get github.com/onsi/ginkgo/v2/ginkgo
-	go get github.com/onsi/gomega/...
+	go get -d github.com/onsi/ginkgo/ginkgo
```
Comment (kaovilai) [inline Makefile:51]: fix deprecation warning

File: controllers/registry.go [hunk 2]
```diff
@@ -761,15 +761,19 @@ func (r *DPAReconciler) parseAzureSecret(secret corev1.Secret, secretKey string)
...
 
-func (r *DPAReconciler) getMatchedKeyValue(matchedKey string, line string) (string, error) {
-	cleanedLine := strings.ReplaceAll(line, " ", "")
-	cleanedLine = strings.ReplaceAll(cleanedLine, "\"", "")
-	matchedKeyValue := strings.Replace(cleanedLine, matchedKey, "", -1)
-	if len(matchedKeyValue) == 0 {
-		r.Log.Info("Could not parse secret for %s", matchedKey)
-		return matchedKeyValue, errors.New("azure secret parsing error")
+// Return value to the right of = sign
+func (r *DPAReconciler) getMatchedKeyValue(key string, s string) (string, error) {
```
Comment (kaovilai) [inline controllers/registry.go:765]: @savitharaghunathan could you double check if changes here is ok for Azure.

Comment (kaovilai) [reply controllers/registry.go:765]: oh wait.. I found a problem..
https://github.com/openshift/oadp-operator/blob/06308a61b1f89d169a1c172626a130799821cc54/controllers/registry.go#L724
brb with fix.

Comment (kaovilai) [reply controllers/registry.go:765]: fixed in a5d575966e150e9843774565e9400b81c13418da

**Review by dymurray — APPROVED** [review]
LGTM at a glance.

File: controllers/registry.go [hunk 3]
```diff
@@ -761,15 +761,19 @@ func (r *DPAReconciler) parseAzureSecret(secret corev1.Secret, secretKey string)
...
+	s = strings.ReplaceAll(s, "\"", "")
```
Comment (shubham-pampattiwar) [inline controllers/registry.go:766]: @kaovilai IMO we should remove the `"` `'`  as well from the string, WDYT ?

Comment (kaovilai) [reply controllers/registry.go:766]: Certainly.

**Review by savitharaghunathan — COMMENTED** [review]
Changes look good to me. Thanks @kaovilai 

/lgtm

**Review by shubham-pampattiwar — APPROVED** [review]
LGTM !

**Pull request merged**